### ISSUE TITLE
Invalid inversePatches when array length changes

### ISSUE THREAD

**Issue by shqld** [unadlib/mutative#64]
Currently, mutative produces invalid inversePatches when array length changes.

test case
```ts
test('#??? inversePatches order', () => {
  checkPatches(
    [0, 1, 2, 3].map((id) => ({ id })),
    (draft) => {
      draft[3].id *= 10;
      draft.splice(0, 1);
    }
  );
});

// (Error in applying inversePatches to nextState)
// Error: Cannot apply patch at '3/id'.
```

patches
```ts
[
  { op: 'replace', path: [ 3, 'id' ], value: 30 },
  { op: 'replace', path: [ 0 ], value: { id: 1 } },
  { op: 'replace', path: [ 1 ], value: { id: 2 } },
  { op: 'replace', path: [ 2 ], value: { id: 30 } },
  { op: 'replace', path: [ 'length' ], value: 3 }
]
```

inversePatches
```ts
[
  { op: 'replace', path: [ 3, 'id' ], value: 30 }, // <- this path does not exist
  { op: 'replace', path: [ 0 ], value: { id: 0 } },
  { op: 'replace', path: [ 1 ], value: { id: 1 } },
  { op: 'replace', path: [ 2 ], value: { id: 2 } },
  { op: 'add', path: [ 3 ], value: { id: 30 } }
]
```

**Comment by unadlib** [comment]
hi @shqld , thank you for reporting this issue. I will fix it later.

**Comment by unadlib** [comment]
hi @shqld , thank you again for reporting this issue. I have released v1.0.11. Feel free to use it.
